python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -640,13 +640,15 @@ def _kill_all_running_actors(self):
 
     def _kill_running_actor(self, actor: ray.actor.ActorHandle):
         """Kill the provided actor and remove it from the pool."""
-        ray.kill(actor)
         del self._num_tasks_in_flight[actor]
+        del self._actor_locations[actor]
+        assert actor not in self._pending_actors
 
     def _kill_pending_actor(self, ready_ref: ray.ObjectRef):
         """Kill the provided pending actor and remove it from the pool."""
         actor = self._pending_actors.pop(ready_ref)
-        ray.kill(actor)
+        assert actor not in self._num_tasks_in_flight
+        assert actor not in self._actor_locations
```

Comment:
Why we're asserting after modified the state?

Reply:
the assert is for a different dict. 
it's to make sure the actor handle is not being ref'ed by any of these 3 dicts